pkg/deploy/controller/deploymentconfig/factory.go

```diff
@@ -193,6 +227,42 @@ func (c *DeploymentConfigController) worker() {
 	}
 }
 
+func (c *DeploymentConfigController) instantiateWorker() {
+	for {
+		if quit := c.instantiate(); quit {
+			return
+		}
+	}
+}
+
+func (c *DeploymentConfigController) instantiate() bool {
+	key, quit := c.instantiateQueue.Get()
+	if quit {
+		return true
+	}
+	defer c.instantiateQueue.Done(key)
+
+	dc, err := c.getByKey(key.(string))
+	if err != nil {
+		utilruntime.HandleError(err)
+	}
+
+	if dc == nil {
+		return false
+	}
+
+	if _, err := c.dn.DeploymentConfigs(dc.Namespace).Instantiate(deployutil.DefaultInstantiateRequestForConfig(dc.Name)); err != nil {
```

Comment:
No, please, we should stop using instantiate inside a controller. Also I was imagining that we could do everything in a single queue.